### ISSUE TITLE
Invalid index

### ISSUE THREAD

**Issue by tpetazzoni** [bootlin/elixir#246]
Bug report from Andrew Lunn:

The index for v6.3 is messed up and return odd results when searching for symbols.

e.g.

https://elixir.bootlin.com/linux/v6.3-rc5/C/ident/of_mdiobus_register_phy

drivers/gpu/drm/drm_framebuffer.c, line 45 (as a function)
drivers/iommu/intel/iommu.c, line 107 (as a function)

Referenced in 3 files:

drivers/gpu/drm/drm_framebuffer.c
line 190
line 225

drivers/iommu/intel/iommu.c
line 290
line 325

drivers/net/mdio/of_mdio.c
line 190
line 225

Only the last two make an sense. And the actual definition and declaration are missing.

**Comment by l1k** [comment]
Hm, are you regenerating the index right now and is that the reason why elixir is down?

My work is grinding to a halt without elixir. :)

**Comment by michaelopdenacker** [comment]
Oops, no. We haven't done anything yet.
Let me have a look...

**Comment by michaelopdenacker** [comment]
One of our servers is down. Our systems administrator will be on it soon.

**Comment by l1k** [comment]
Thank you so much, it's working again. My apologies for hijacking this issue.

**Comment by safinaskar** [comment]
I still see broken links. https://elixir.bootlin.com/linux/v6.3/C/ident/path_mount has a link to https://elixir.bootlin.com/linux/v6.3/source/arch/x86/kvm/cpuid.c#L3249 , but there is no line 3249 there

**Comment by michaelopdenacker** [comment]
It seems that we're having a database corruption here, as I don't get the issue when I index a simplified tree containing only the v6.3 tag. I'm regenerating a new database to see whether this helps...
I'll keep you posted, of course!

**Comment by andy-shev** [comment]
Not sure if it's done, but for me results are still odd:
https://elixir.bootlin.com/linux/latest/A/ident/bsw_pwm_lookup
(look into tests, which have nothing to do with the searched symbol).

**Comment by michaelopdenacker** [comment]
Right, and that's why I wonder whether we have a corrupted database.

**Comment by tpetazzoni** [comment]
Even if it's a corrupted database, it would be nice to understand *why* it got corrupted. Indeed, it's not great for the on-going maintenance of Elixir to rely on having to randomly rebuild the database to fix weird issues.

**Comment by andy-shev** [comment]
v6.4 database is definitely corrupted. On top of that (dunno if it's a separate bug) the enum fields are not indexed. Ex.: https://elixir.bootlin.com/linux/v6.4-rc2/C/ident/pin_config_param.

**Comment by michaelopdenacker** [comment]
The database corruption is gone now, after deploying a freshly generated database.
It's true I don't know why the database got corrupted, but it could be a pretty long investigation. Maybe the server was rebooted while the database was getting updated. From the date that was first reported, which matches when the server was rebooted, this could be a possibility. I'd suggest to see whether this happens again before investing further time on this.
Thanks again for reporting this!
